cashu/lightning/lnbits.py

```diff
@@ -60,37 +63,48 @@ async def create_invoice(
 
         data["memo"] = memo or ""
         try:
-            r = self.s.post(url=f"{self.endpoint}/api/v1/payments", json=data)
+            r = await self.s.post(url=f"{self.endpoint}/api/v1/payments", json=data)
+        except:
+            return InvoiceResponse(False, None, None, "Request error.")
+        try:
             r.raise_for_status()
         except:
-            return InvoiceResponse(False, None, None, r.json()["detail"])
+            data = await r.json()
+            return InvoiceResponse(False, None, None, data["detail"])
```

Comment:
Separated into two different `try/except` blocks as the first can fail because of network problems, the second is for reponse problems.

Also, `data = await r.json()` depends on `r` being a valid response, so we can't access it in the exception handler unless the initial request succeeded to some extent (it may still throw "this isn't JSON!" error if the response was something like a Nginx 502 bad gateway, but that was the behaviour of the original code too)